src/Container/Lifetime/LifetimeContainer.cs

```diff
@@ -100,12 +100,29 @@ protected virtual void Dispose(bool disposing)
             {
                 lock (_items)
                 {
+                    var exceptions = new List<Exception>();
                     foreach (var disposable in _items.OfType<IDisposable>().Reverse())
                     {
-                        disposable.Dispose();
+                        try
+                        {
+                            disposable.Dispose();
+                        }
+                        catch (Exception e)
+                        {
+                            exceptions.Add(e);
+                        }
                     }
 
                     _items.Clear();
+
+                    if (exceptions.Count == 1)
+                    {
+                        throw exceptions.First();
```

Comment:
Shouldn't the exception returned be consistent?

Reply:
What I mean is, shouldn't it always be an AggregateException, then the consumer can flatten

Reply:
The way you do it is fine

Reply:
@jasonp123  For best backward compatibility,  when there's single exception, the type of exception being thrown remains the same before and after this change